hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsRegionServerWrapperImpl.java

```diff
@@ -115,6 +118,8 @@
   private volatile long mobFileCacheCount = 0;
   private volatile long blockedRequestsCount = 0L;
   private volatile long averageRegionSize = 0L;
+  protected volatile Map<String, ArrayList<Long>> requestsCountCache = new
```

Comment:
You're not changing the object that `requestsCountCache` points to, are you? I think you can drop `volatile` and make this `final`.